app/views/users/phone_setup/index.html.erb

```diff
@@ -23,7 +23,16 @@
       url: phone_setup_path,
     ) do |f| %>
 
-  <%= render PhoneInputComponent.new(form: f, confirmed_phone: false, required: true, class: 'margin-bottom-4') %>
+  <%= render PhoneInputComponent.new(
+        form: f,
+        confirmed_phone: false,
+        required: true,
+        captcha_submit_button_id: 'submit_phone_form_button',
```

Comment:
should we pull this ID into a constant or a local variable so we can help make sure it stays consistent with the button instantiation below?

Reply:
Hm, I see what you mean. I already wasn't very fond of how this ended up with "linking" the two components (I'd even gone as far as wondering whether to just remove the "bypass" logic and let reCAPTCHA always run). I think extracting the ID would help keep it consistent, but also further highlight the fragility of linking this way.

I'll try to think through some options here.

Reply:
In other cases, we wrap components in each other, is this approach because the two things are in separate parts of the DOM? what about an approach like this:

```erb
<%= render PhoneCapchaForm.new do |p| %>
  <%= render p.phone_input %>
  ...
  ...
  <%= render p.captcha_submit %>
<% end %>
```

Reply:
Ooh, yes, I think we could do something like that! Or at least something where we create a block-able version of  `CaptchaSubmitButtonComponent` that exposes an internally-generated ID, like...

```erb
<%= render CaptchaSubmitButtonComponent.new do |c| %>
  <%= render PhoneInputComponent.new(captcha_submit_button_id: c.id) %>

  <%# ... %>

  <%= c.button %>
<% end %>
```

Reply:
I started exploring this, but it got extra tricky with dealing with optional slots for the component. I went a different route in b0e7b10ac, which arguably leans further into implicit association, but at the benefit of simpler code and absence of ID association.